app/controllers/graders_controller.rb

```diff
@@ -122,11 +122,18 @@ def global_actions
             found_empty_submission = true
           end
         end
-        if found_empty_submission
-          randomly_assign_graders(filtered_grouping_ids, grader_ids)
-          flash_now(:info, I18n.t('graders.group_submission_no_files'))
-        else
-          randomly_assign_graders(grouping_ids, grader_ids)
+        begin
+          weights = params[:weightings].map { |weight| Float(weight) }
+          if weights.reduce(:+) == 0 || weights.select(&:negative?).length > 0
+            head :bad_request
```

Comment:
Whenever you use a `head` or `render` method, it should immediately be followed by a `return` to stop the rest of the function from executing. Otherwise we open ourselves up to getting "double render" errors.

Also, please flash an error message here describing the problem.